### PR TITLE
Create a basic wiki

### PR DESCRIPTION
**Pull request by daveynight**
Addresses #182.

WIP

### CODE REVIEW

**Comment by ghost** [comment]
Just thinking if we should really have explicit article sections like that. I guess it can't hurt but it sure does seem like overkill in preference to relying on bbcode/markdown.

**Comment by ghost** [comment]
@ajax-wcd Having explicit articles isn't completely necessary, but it would make storing articles easier with the new pre-defined template system, as well as creating the TOC based on the template.

**Comment by daveynight** [comment]
It seemed like a simple way to enforce some of the formatting constraints: 1 picture/section, each section gets a title. I also like how it this works with autoform (such as in http://autoform.meteor.com/qfdetails) but I could change this to rely more on markdown if that's a better solution.

File: packages/meteor-gazelle-wiki/lib/methods.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Meteor.methods({
+  'wiki/createArticle': function (doc) {
+    var article = new Article();
+    article.set('title', doc.title);
+    article.set('body', doc.content);
+    article.set('createdBy', this.userId);
+    article.set('locked', false);
+    article.save();
+  },
+  'wiki/lockArticle': function (article) {
```
Comment (jimrollenhagen) [inline packages/meteor-gazelle-wiki/lib/methods.js:12]: Is locking articles a thing?

File: packages/meteor-gazelle-wiki/lib/publications.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Meteor.publish('wiki', function () {
+  return Articles.find({}, { sort: { createdat: -1 }});
```
Comment (jimrollenhagen) [inline packages/meteor-gazelle-wiki/lib/publications.js:2]: s/createdat/createdAt/

File: packages/meteor-gazelle-wiki/lib/routes.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+WikiController = ApplicationController.extend({
+  waitOn: function () {
+    return Meteor.subscribe('announcements');
+  },
+  data: function () {
+    return {
+      wiki: Articles.find({}, { sort: { createdAt: -1 }})
```
Comment (jimrollenhagen) [inline packages/meteor-gazelle-wiki/lib/routes.js:7]: Probably don't want to send all articles to the client at once, but we can come back to this later.

File: packages/meteor-gazelle-wiki/lib/templates/article.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<template name="article">
+  {{ title }}
+  {{ sections }}
+  {{ aliases }}
+  {{ createdAt }}
+  {{ updatedAt }}
+</template>
```
Comment (jimrollenhagen) [inline packages/meteor-gazelle-wiki/lib/templates/article.html:7]: I have a feeling there's more to come here :)

**Comment by jimrollenhagen** [comment]
Tests?

File: packages/meteor-gazelle-wiki/lib/methods.js (hunk 1 above)
Comment (ghost) [inline packages/meteor-gazelle-wiki/lib/methods.js:12]: I don't know if it's a thing currently but it should be for certain articles. 

I wouldn't make a single function just for locking articles though. From a usage perspective the user would probably need to hit edit on the article first which would then expose the locked checkbox. Hitting save would then call the article edit method.

**Comment by daveynight** [comment]
I'm hesitant to write more tests before we resolve the conflict TinyTest has been having with Validations.

File: packages/meteor-gazelle-wiki/lib/templates/article.html (hunk 4 above)
Comment (daveynight) [inline packages/meteor-gazelle-wiki/lib/templates/article.html:7]: Certainly. But routing isn't working in our app, yet, (and I've tried and failed to fix it,) so I'm unable to work more on the templates until it is.

File: packages/meteor-gazelle-wiki/lib/routes.js (hunk 3 above)
Comment (daveynight) [inline packages/meteor-gazelle-wiki/lib/routes.js:7]: Heh, sure. I've already removed this data field and put a more appropriate one into each of the router 'actions' in the `ApplicationController`.

Incidentally, the `ApplicationController` isn't currently being made available through `api.export()` from the `core` package.

File: packages/meteor-gazelle-wiki/lib/publications.js (hunk 2 above)
Comment (daveynight) [inline packages/meteor-gazelle-wiki/lib/publications.js:2]: √

File: packages/meteor-gazelle-wiki/lib/methods.js (hunk 1 above)
Comment (daveynight) [inline packages/meteor-gazelle-wiki/lib/methods.js:12]: Understood.

**Comment by rhomes-wcd** [comment]
Hey Davey, I fixed this issue in the most recent merge to master. It was stemming from missing package versions in each package.js file. Also, with that same merge, we are moving to the Velocity/Jasmine testing frameworks. If you rebase from master, refactor your tests, they should work just fine. You can ping me in #gazelle if you have any questions.

**Comment by daveynight** [comment]
O RLY! Cool :) Thanx

File: packages/meteor-gazelle-wiki/lib/methods.js (hunk 1 above)
Comment (thomwiggers) [inline packages/meteor-gazelle-wiki/lib/methods.js:12]: @ajax-wcd why actually should it be a thing? If an article should be locked its editing permissions should just be put higher.

**Comment by ghost** [comment]
Closing this.

**Pull request closed without merging**